deprecate interactive API

vrancurel opened the pull request:
- Move the interactive_api examples under `openfl-tutorials/deprecated`
- Mark interactive_api as deprecated in the doc
- Remove interactive_api from CI/CD

teoparvanov commented on openfl-tutorials/Federated_Pytorch_MNIST_Tutorial.ipynb:
This file doesn't seem related to the Interactive API deprecation. You may have included it by mistake?

vrancurel replied:
Correct, removed from commit

teoparvanov commented on .github/workflows/ubuntu.yml:
```diff
@@ -45,28 +45,8 @@ jobs:
-  
   cli:
-    needs: [lint, pytest-coverage, interactive-kvasir]
+    needs: [lint, pytest-coverage]
```
I'd suggest including `workflow_interface_101_mnist` and `taskrunner` here - to highlight the shift towards Workflow API and TaskRunner API. 

WDYT?

vrancurel replied:
I will do another commit to reintroduce an interactive workflow example from new worflow API

teoparvanov reviewed (commented):
LGTM at first glance. Here are a couple of things I noticed:

teoparvanov commented on .github/workflows/windows.yml:
```diff
@@ -27,28 +27,8 @@ jobs:
-  
   cli: # from taskrunner.yml
-    needs: [pytest-coverage, interactive-kvasir]
+    needs: [pytest-coverage]
```
Same here, by the way - suggesting to run `workflow_interface_101_mnist` and `taskrunner` instead.

vrancurel replied:
Same here

kminhta commented on docs/about/releases.md:
```diff
@@ -11,7 +11,7 @@
 - **Federated Evaluation**: Federated evaluation allows for the assessment of ML models in a federated learning system by validating the model's performance locally on decentralized collaborator nodes, and then aggregating these metrics to gauge overall effectiveness, without compromising data privacy and security. FE is now officially supported by OpenFL, including [example tutorials](https://op ...
 
-- **Expanded AI Accelerator Support**:  Intel® Data Center GPU Max Series support via the Intel® Extension for PyTorch, including examples for training on datasets such as [MNIST](https://github.com/securefederatedai/openfl/blob/develop/openfl-tutorials/experimental/104_MNIST_XPU.ipynb) (via Workflow API) and [TinyImageNet](https://github.com/securefederatedai/openfl/tree/develop/openfl-tutorials ...
+- **Expanded AI Accelerator Support**:  Intel® Data Center GPU Max Series support via the Intel® Extension for PyTorch, including examples for training on datasets such as [MNIST](https://github.com/securefederatedai/openfl/blob/develop/openfl-tutorials/experimental/104_MNIST_XPU.ipynb) (via Workflow API) and [TinyImageNet](https://github.com/securefederatedai/openfl-contrib/tree/main/openfl_cont ...
```
From my comment on the [openfl-contrib PR](https://github.com/securefederatedai/openfl-contrib/pull/9#issuecomment-2463389082)

Would it be better to move these to a subfolder `openfl-tutorials/deprecated/` for v1.7 then remove outright in v1.6 instead of to the contrib repo?

teoparvanov replied:
This is the suggestion indeed, based on our offline discussion.

kta-intel commented:
Thanks @vrancurel, this is definitely a much needed PR! However, I'd just like some clarity on the plan

Commit:
deprecate interactive API

- Point interactive_api examples to openfl-contrib (where they have been moved)
- Mark interactive_api as deprecated in the doc
- Remove interactive_api from CI/CD

Signed-off-by: Rancurel, Vianney <vianney.rancurel@intel.com>

vrancurel commented:
Thanks @tanwarsh please merge as needed.

teoparvanov commented:
Hi @vrancurel , do you intend to refactor `kvasir-interactive` to Workflow API in this PR, or a separate one?
If done separately, then this PR could be merged as-is. So, let me know.

vrancurel commented:
In a separate one, please merge @teoparvanov
